plutus-contract/src/Plutus/Contract/Request.hs

```diff
@@ -214,21 +216,31 @@ isSlot ::
 isSlot = Promise . awaitSlot
 
 -- | Get the current slot number
-{-# DEPRECATED currentSlot "It was renamed to 'currentPABSlot', this function will be removed" #-}
+{-# DEPRECATED currentSlot "Use currentNodeClientSlot instead" #-}
 currentSlot ::
     forall w s e.
     ( AsContractError e
     )
     => Contract w s e Slot
 currentSlot = currentPABSlot
```

Comment:
Do we have to remove this one? It's a nice and short name to get the current slot, and we kinda know it's the client node's one (we can still precise it in the doc if needed)

Reply:
Honestly, I prefer explicit naming.

The name `currentPABSlot` is ambiguous as it can refer to the slot that is currently indexed by PAB, or the slot of the local running node. In the current implementation, it returns the slot of the local running, hence why I changed the name to make it more clear. If we really need the slot that PAB has currently indexed, then yes we can create `currentPABSlot`, but there is not actual use case for that.

Also, PAB is too specific to the `plutus-pab` package. And since it's future is unclear, might as well change the name now.

Reply:
`currentPABSlot` is bad, I was advocating for `currentSlot` here. But if you think it can be misleading, we can deprecate it, no problem. :)

Reply:
Oh! We actually used to have `currentSlot` and end-users were confused as to what that refers to (PAB? chain-index? node client?).

Reply:
Ok, so it's preferable to deprecate it in this case.